src/main/java/edu/wpi/grip/core/SocketHint.java

```diff
@@ -27,13 +28,25 @@
      *                             can consist of two elements that correspond to a minimum and maximum value.  The
      *                             property does not make sense for all types and is left unspecified for some.
      * @param initialValueSupplier A function that returns an initial value for newly-created sockets.
+     * @param publishable          If this isn't true, the socket won't be able to be set as published, and the GUI
+     *                             should also not show a button to publish it.  This is false by default.
```

Comment:
Most things probably won't be intended to be publishable, so it's more convenient to not have to use the really long constructor